From bbc525865dda18ab1bd9f75eb2a33fbe3f602f29 Mon Sep 17 00:00:00 2001
From: Flynn xu <flynn.xu@nxp.com>
Date: Thu, 15 Nov 2018 16:30:24 +0800
Subject: [PATCH 5133/5242] MLK-20380: i2c: xenfront: fix i2cdetect bug

commit  f210076aade6f5b6a5544d66108f292000a77305 from
https://source.codeaurora.org/external/imx/linux-imx.git

The bug is when using i2cdetect in DomU, some i2c device will be
skiped, fixed this through add I2C_FUNC_SMBUS_QUICK functionality.

Signed-off-by: Flynn xu <flynn.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/i2c/busses/xen-i2cfront.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/xen-i2cfront.c b/drivers/i2c/busses/xen-i2cfront.c
index b041415..8a18a79 100644
--- a/drivers/i2c/busses/xen-i2cfront.c
+++ b/drivers/i2c/busses/xen-i2cfront.c
@@ -262,7 +262,7 @@ static int i2cfront_smbus_xfer(struct i2c_adapter *adapter, u16 addr,
 static u32 i2cfront_func(struct i2c_adapter *adapter)
 {
 	return I2C_FUNC_I2C | I2C_FUNC_SMBUS_BYTE_DATA |
-			I2C_FUNC_SMBUS_BYTE | I2C_FUNC_SMBUS_WORD_DATA |
+			I2C_FUNC_SMBUS_BYTE | I2C_FUNC_SMBUS_WORD_DATA | I2C_FUNC_SMBUS_QUICK |
 			I2C_FUNC_SMBUS_BLOCK_DATA | I2C_FUNC_SMBUS_I2C_BLOCK;
 }
 
-- 
1.7.9.5

